qiskit/pulse/macros.py

```diff
@@ -104,3 +189,21 @@ def measure_all(backend) -> Schedule:
         A schedule corresponding to the inputs provided.
     """
     return measure(qubits=list(range(backend.configuration().n_qubits)), backend=backend)
+
+
+def combine_schedules(qubits: List, schedule: Schedule, default_schedule: Union[Schedule, ScheduleBlock],measure_group_qubits: List, qubit_mem_slots: Optional[Dict[int, int]] = None):
```

Comment:
I think this should be a function `remap_memory_slot(schedule: Schedule, qubit_mem_slots: Dict[int, int]) -> Schedule` because it seems just replacing the slot index. @TsafrirA does it help creg index matching? Let us know if you have other ideas.

Reply:
I change the code at https://github.com/Qiskit/qiskit-terra/pull/9912/commits/509392a57b805f64e7dd4030d3c6a109a64427f5